cosmo_tester/test_suites/test_multiple_managers/test_multiple_managers.py

```diff
@@ -105,12 +182,13 @@ def test_create_snapshot_and_restore_on_another_manager(self):
         self.additional_clients[0].snapshots.upload(snapshot_file_path, 'node')
         # creating a snapshots is asynchronous, but it lasts a second or two...
         time.sleep(3)
-        try:
-            os.remove(snapshot_file_path)
-        except OSError:
-            pass
         self.logger.info('Snapshot uploaded.')
 
+        self.logger.info('Removing snapshot file...')
+        if os.path.isfile(snapshot_file_path):
+            os.remove(snapshot_file_path)
+        self.logger.info('Snapshot file removed.')
+
         self.logger.info('Restoring snapshot...')
         self._restore_snapshot(self.additional_clients[0], 'node')
         self.logger.info('Snapshot restored (there is a chance, actually).')
```

Comment:
Maybe check if the executions state is appropriate (no "there is a chance" please ;) ) I do it in my test